### PR TITLE
Fix compilation pipeline for IR tests.

### LINKED ISSUE #695: [test_core.py::test_convert2d] fails to run correctly on PVC
To reproduce, remove:
```
    if is_xpu() and M == 128 and N == 128 and interm_layout and (dst_layout.sz_per_thread == [1, 8]
                                                                 or dst_layout.sz_per_thread == [4, 4]):
        pytest.skip("FIXME: out of resource: shared memory")
```
and run:
`python3 -m pytest --verbose --device xpu python/test/unit/language/test_core.py::test_convert2d`

### PR DESCRIPTION
**Pull request by ienkovich**
Resolves #695

There is still one test_convert2d configuration that fails but looks like the problem is in IGC, see #729

### CODE REVIEW

**Review by etiotto — DISMISSED** [review]

**Review by whitneywhtsang — DISMISSED** [review]

**Comment by ienkovich** [comment]
Looks like our compilation pipeline is required for some IR tests, there must be some Intel-specific lowering in our Triton->TritonGPU pipeline that is essential for those tests. I'll keep working on that.

**Comment by whitneywhtsang** [comment]
> Looks like our compilation pipeline is required for some IR tests, there must be some Intel-specific lowering in our Triton->TritonGPU pipeline that is essential for those tests. I'll keep working on that.

Should be related to https://github.com/intel/intel-xpu-backend-for-triton/issues/588.

**Comment by ienkovich** [comment]
> Should be related to #588.

Thanks! This is blocked by #588 then.

**Comment by ienkovich** [comment]
@whitneywhtsang @etiotto 
I propose to merge the modified patch version where we skip failing tests and then create new issues to work on their enabling. Currently, all tests written on TritonGPU MLIR go through the pipeline that optimizes out layout conversions. It means we don't adequately test layout conversions. Failures revealed by the compilation pipeline fix are not DPAS-related only, `test_reduce_layouts` also has failures on blocked layout. Keeping the wrong compilation pipeline, we are risking to introduce new regressions (e.g. #779 introduced a new problem for DPAS conversion that couldn't be found by CI and we might similarly break something for other layouts).

**Comment by ienkovich** [comment]
@etiotto 
Which regressions do we need to fix before pushing this one?

**Comment by etiotto** [comment]
> @etiotto Which regressions do we need to fix before pushing this one?

I believe fixing #588 would allow this PR to land.

**Comment by etiotto** [comment]
Created https://github.com/intel/intel-xpu-backend-for-triton/issues/793 to fix tests (skipped by this PR) that involve DPAS layout conversions.

File: python/test/unit/language/test_core.py
Comment (whitneywhtsang) [inline python/test/unit/language/test_core.py:1]: Can we please use `pytest.skip("FIXME:...)`? I search all skips added by us using that string.

Comment (ienkovich) [reply python/test/unit/language/test_core.py:1]: Done

**Review by etiotto — APPROVED** [review]

**Review by whitneywhtsang — APPROVED** [review]

**Comment by ienkovich** [comment]
Created issues for non-DPAS failures: #794 #795

**Pull request merged**